dash-pipeline/bmv2/dash_outbound.p4

```diff
@@ -5,212 +5,197 @@
-    }
-
-    action route_direct(bit<1> meter_policy_en,
-                        bit<16> meter_class) {
-        set_route_meter_attrs(meter_policy_en, meter_class);
-        /* send to underlay router without any encap */
-    }
-
     action drop() {
-        meta.dropped = true;
+        meta.pkt_meta.dropped = true;
     }
 
-    action route_service_tunnel(bit<1> is_overlay_dip_v4_or_v6,
-                                IPv4ORv6Address overlay_dip,
-                                bit<1> is_overlay_dip_mask_v4_or_v6,
-                                IPv4ORv6Address overlay_dip_mask,
-                                bit<1> is_overlay_sip_v4_or_v6,
-                                IPv4ORv6Address overlay_sip,
-                                bit<1> is_overlay_sip_mask_v4_or_v6,
-                                IPv4ORv6Address overlay_sip_mask,
-                                bit<1> is_underlay_dip_v4_or_v6,
-                                IPv4ORv6Address underlay_dip,
-                                bit<1> is_underlay_sip_v4_or_v6,
-                                IPv4ORv6Address underlay_sip,
-                                dash_encapsulation_t dash_encapsulation,
-                                bit<24> tunnel_key,
-                                bit<1> meter_policy_en,
-                                bit<16> meter_class) {
-        /* Assume the overlay addresses provided are always IPv6 and the original are IPv4 */
-        /* assert(is_overlay_dip_v4_or_v6 == 1 && is_overlay_sip_v4_or_v6 == 1);
-        assert(is_overlay_dip_mask_v4_or_v6 == 1 && is_overlay_sip_mask_v4_or_v6 == 1);
-        assert(is_underlay_dip_v4_or_v6 != 1 && is_underlay_sip_v4_or_v6 != 1); */
-        meta.encap_data.original_overlay_dip = hdr.ipv4.src_addr;
-        meta.encap_data.original_overlay_sip = hdr.ipv4.dst_addr;
-
-        service_tunnel_encode(hdr,
-                              overlay_dip,
-                              overlay_dip_mask,
-                              overlay_sip,
-                              overlay_sip_mask);
-
-        /* encapsulation will be done in apply block based on dash_encapsulation */
-        meta.encap_data.underlay_dip = underlay_dip == 0 ? meta.encap_data.original_overlay_dip : (IPv4Address)underlay_dip;
-        meta.encap_data.underlay_sip = underlay_sip == 0 ? meta.encap_data.original_overlay_sip : (IPv4Address)underlay_sip;
-        meta.encap_data.overlay_dmac = hdr.ethernet.dst_addr;
-        meta.encap_data.dash_encapsulation = dash_encapsulation;
-        meta.encap_data.service_tunnel_key = tunnel_key;
-        set_route_meter_attrs(meter_policy_en, meter_class);
+    action outbound_metadata_publish(dash_match_stage_t next_stage,
+                                     dash_routing_type_t routing_type,
+
+                                     /* keys for routing */
+                                     @SaiVal[type="sai_object_id_t"] dash_oid_t pipeline_oid,
```

Comment:
please run make sai-headers and avoid changes to existing field names, e.g., use eni_id for pipeline_id, vnet_id for mapping_id.